contract/src/primitives.rs

```diff
@@ -0,0 +1,197 @@
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::serde::{Deserialize, Serialize};
+use near_sdk::{AccountId, PublicKey};
+use std::collections::{BTreeMap, HashSet};
+
+pub mod hpke {
```

Comment:
What is `hpke`?

Reply:
It's used to determine the public key type used for encrypting messages.
Crate: https://docs.rs/crate/hpke/latest
(it is not added by this PR)